src/components/data-table/ha-data-table.ts

```diff
@@ -627,7 +627,7 @@ export class HaDataTable extends LitElement {
           border-top: 1px solid var(--divider-color);
         }
 
-        .mdc-data-table__row:not(.mdc-data-table__row--selected):hover {
+        .mdc-data-table__row:is(.clickable):not(.mdc-data-table__row--selected):hover {
```

Comment:
```suggestion
        .mdc-data-table__row.clickable:not(.mdc-data-table__row--selected):hover {
```

Reply:
Good suggestion. This was an oversight on my part, sorry.